src/elements/content-sidebar/additional-tabs/AdditionalTabs.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * @flow
+ * @file Preview sidebar additional tabs components
+ * @author Box
+ */
+
+import React, { PureComponent } from 'react';
+import AdditionalTab from './AdditionalTab';
+import AdditionalTabsLoading from './AdditionalTabsLoading';
+import './AdditionalTabs.scss';
+
+type Props = {
+    tabs: ?AdditionalSidebarTabs,
+};
+
+type State = {
+    isLoading: boolean,
+};
+
+class AdditionalTabs extends PureComponent<Props, State> {
+    initialState: State = {
+        isLoading: true,
+    };
+
+    constructor(props) {
+        super(props);
+
+        this.numLoadedTabs = 0;
+
+        this.state = { ...this.initialState };
+    }
+
+    onImageLoad = () => {
+        const { tabs } = this.props;
+
+        this.numLoadedTabs += 1;
+
+        // If we're displaying the more tab, consider it loaded.
+        const hasMoreTab = tabs.find(tab => tab.id < 0 && !tab.iconUrl);
```

Comment:
does this need to be done on each `onImageLoad` invocation or can it be done once?

Reply:
We only need to do it once, but we need to wait until we're passed in the actual tab data, which doesn't happen on init.